src/voter/mod.rs

```diff
@@ -361,6 +368,51 @@ type FinalizedNotification<H, N, E> = (
 	Commit<H, N, <E as Environment<H, N>>::Signature, <E as Environment<H, N>>::Id>,
 );
 
+// instantiates the last round, to be backgrounded until its estimate is finalized.
+//
+// this round must be comletable based on the passed votes (and if not, `None` will be returned)
+// but it may be the case that there are some more votes to propagate in order to push
+// the estimate backwards.
+fn instantiate_last_round<H, N, E: Environment<H, N>>(
+	voters: VoterSet<E::Id>,
+	last_round_votes: Vec<SignedMessage<H, N, E::Signature, E::Id>>,
+	last_round_number: u64,
+	last_round_base: (H, N),
+	finalized_sender: mpsc::UnboundedSender<FinalizedNotification<H, N, E>>,
+	env: Arc<E>,
+) -> Option<VotingRound<H, N, E>> where
+	H: Hash + Clone + Eq + Ord + ::std::fmt::Debug,
+	N: Copy + BlockNumberOps + ::std::fmt::Debug,
+{
+	let last_round_tracker = crate::round::Round::new(crate::round::RoundParams {
+		voters: voters,
+		base: last_round_base,
+		round_number: last_round_number,
+	});
+
+	let last_round_parent_state = env.round_parent_state(last_round_number)
```

Comment:
Do we actually need this? The previous round state is only necessary for voting and to make sure we don't start `R + 1` while the estimate for `R - 1` hasn't been finalized. In this case we are spawning `R - 1` and `R`, we won't vote on `R - 1` (we must have done so already previously otherwise it wouldn't be completed) and we only need to make sure that `R` has access to `R - 1`'s state to track any estimate change.

Reply:
I think i tried it without; the problem is that the round concludes immediately when the grandparent state is `None`.